src/mailingui/components/buttons/ReactEmailButtonFork.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from "react";
+
+type ButtonElement = React.ElementRef<"a">;
+type RootProps = React.ComponentPropsWithoutRef<"a">;
+
+export interface ButtonProps extends RootProps {
+  pX?: number;
+  pY?: number;
+}
+
+export const Button = React.forwardRef<ButtonElement, Readonly<ButtonProps>>(
```

Comment:
I'm curious, why would we need this utility for component props? Readonly constructs a type which properties cannot be reassigned. Is this a pattern we need?